rust/lance/src/dataset/write.rs

```diff
@@ -554,6 +569,37 @@ impl WriterGenerator {
     }
 }
 
+// Given input options resolve what the commit handler should be.
+async fn resolve_commit_handler(
```

Comment:
Why did this method show up in here?

Reply:
It was in `src/dataset.rs` and is used both in `CommitBuilder` and `InsertBuilder`. I thought about moving it to `lance-io`, but the logic ended up feeling specific to how those builder APIs wanted to interpret handle parameters.